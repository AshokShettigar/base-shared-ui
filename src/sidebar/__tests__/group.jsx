import { shallow } from 'enzyme';
import { SidebarGroup } from 'platform-common-ui';

describe('sidebar/group', function() {
    it('should render the sidebar group', function() {
        const wrapper = shallow(
            <SidebarGroup onChangeExpandCollapse={() => true} title="Group1">
                <div>Item 1</div>
            </SidebarGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the sidebar group with collapsed state', function() {
        const wrapper = shallow(
            <SidebarGroup isCollapsed onChangeExpandCollapse={() => true} title="Group1">
                <div>Item 1</div>
            </SidebarGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the sidebar group with custom class names', function() {
        const wrapper = shallow(
            <SidebarGroup
                className="my-class"
                onChangeExpandCollapse={() => true}
                title="Group1"
                titleClassName="my-title=class"
            >
                <div>Item 1</div>
            </SidebarGroup>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should call onChangeExpandCollapse function when title is clicked', function() {
        const handler = jest.fn();
        const wrapper = shallow(
            <SidebarGroup onChangeExpandCollapse={handler} title="Group1">
                <div>Item 1</div>
            </SidebarGroup>
        );
        wrapper.find('.cn-sidebar-group__title').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
    });
});
