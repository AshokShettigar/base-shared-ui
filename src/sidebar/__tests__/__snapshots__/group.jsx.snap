// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar/group should render the sidebar group 1`] = `
<div
  className="cn-sidebar-group"
  data-component="Group"
>
  <span
    className="cn-sidebar-group__title"
    onClick={[Function]}
    role="button"
  >
    Group1
  </span>
  <ul
    className="cn-sidebar-group__list"
  >
    <div>
      Item 1
    </div>
  </ul>
</div>
`;

exports[`sidebar/group should render the sidebar group with collapsed state 1`] = `
<div
  className="cn-sidebar-group"
  data-component="Group"
>
  <span
    className="cn-sidebar-group__title"
    onClick={[Function]}
    role="button"
  >
    Group1
  </span>
</div>
`;

exports[`sidebar/group should render the sidebar group with custom class names 1`] = `
<div
  className="cn-sidebar-group my-class"
  data-component="Group"
>
  <span
    className="cn-sidebar-group__title my-title=class"
    onClick={[Function]}
    role="button"
  >
    Group1
  </span>
  <ul
    className="cn-sidebar-group__list"
  >
    <div>
      Item 1
    </div>
  </ul>
</div>
`;
